lib/types/signers_list.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Kadena.Types.SignersList do
+  @moduledoc """
+  `SignersList` struct definition.
+  """
+  alias Kadena.Types.Signer
+
+  @behaviour Kadena.Types.Spec
+
+  @type signer :: Signer.t()
+  @type signers :: list(signer())
+
+  @type t :: %__MODULE__{list: signers()}
+
+  defstruct list: []
+
+  @impl true
+  def new(signers), do: build_list(%__MODULE__{}, signers)
+
+  @spec build_list(list :: t(), signers :: signers()) :: t() | {:error, Keyword.t()}
+  defp build_list(list, []), do: list
+
+  defp build_list(%__MODULE__{list: list}, [value | rest]) do
+    with %Signer{} = signer <- Signer.new(value) do
+      build_list(%__MODULE__{list: [signer | list]}, rest)
+    end
```

Comment:
@CristhianRodriguezMolina @santiagoboterokommit A better approach would be to use a case instead of a with